proto/kava/auction/v1beta1/query.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+package kava.auction.v1beta1;
+
+import "kava/auction/v1beta1/genesis.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/kava-labs/kava/x/auction/types";
+
+// Query defines the gRPC querier service for auction module
+service Query {
+  // Params queries all parameters of the auction module.
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/auction/params";
+  }
+
+  // Auction queries an individual Auction by auction ID
+  rpc Auction(QueryAuctionRequest) returns (QueryAuctionResponse) {
+    option (google.api.http).get = "/auction/auction";
+  }
+
+  // Auctions queries auctions filtered by asset denom, owner address, phase, and auction type
+  rpc Auctions(QueryAuctionsRequest) returns (QueryAuctionsResponse) {
+    option (google.api.http).get = "/auction/auctions";
+  }
+
+  // NextAuctionID queries the next auction ID
+  rpc NextAuctionID(QueryNextAuctionIDRequest) returns (QueryNextAuctionIDResponse) {
+    option (google.api.http).get = "/auction/next-auction-id";
+  }
+}
+
+// QueryParamsRequest defines the request type for querying x/auction parameters.
+message QueryParamsRequest {}
+
+// QueryParamsResponse defines the response type for querying x/auction parameters.
+message QueryParamsResponse {
+  Params params = 1 [(gogoproto.nullable) = false];
+}
+
+// QueryAuctionRequest is the request type for the Query/Auction RPC method.
+message QueryAuctionRequest {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string auction_id = 1;
```

Comment:
Any reason for string vs uint64?

Reply:
https://github.com/Kava-Labs/kava/pull/1052/commits/e3c0f253bd10ae79005926603c17e065303e1eb3